spec/menubar/utils_spec.lua

```diff
@@ -65,6 +65,64 @@ describe("menubar.utils lookup_icon_uncached", function()
         assert_found_in_path('awesome',  '/.icons/awesome/64x64/apps/awesome.png')
         assert_found_in_path('awesome2', '/.icons/awesome/scalable/apps/awesome2.png')
     end)
+
+-- Check for no argument :: if not icon_file or icon_file == "" then
+    it('no icon specified', function()
+        assert.is_false(utils.lookup_icon_uncached())
+        assert.is_false(utils.lookup_icon_uncached(nil)) -- Same as above?
+        assert.is_false(utils.lookup_icon_uncached(false))
+        assert.is_false(utils.lookup_icon_uncached(''))
+    end)
+
+-- Check for icon not in search path :: if icon_file:sub(1, 1) == '/' and supported_icon_formats[icon_file_ext] then
+    it('finds icons even those not in the search paths when full path specified', function()
+
+        -- Shimmed icon base directories contain the following icons:
+        --
+        --     usr/share/icon5.png
+        --     usr/share/icon6.xpm
+        --     usr/share/icon7.svg
+
+--        assert_found_in_path('/usr/share/icon5.png', '/usr/share/icon5.png')
+--        assert_found_in_path('/usr/share/icon6.xpm', '/usr/share/icon6.xpm')
+--        assert_found_in_path('/usr/share/icon7.svg', '/usr/share/icon7.svg')
+
+        assert.is_same(nil, utils.lookup_icon_uncached('/.png')) -- [For Uli] supported file does not exist in location
+        assert.is_same(nil, utils.lookup_icon_uncached('/blah/icon6.png')) -- supported file does not exist in location
+    end)
+
+-- Check icon with specified extension matching :: if supported_icon_formats[icon_file_ext] and
+    it('finds icons with specified supported extension in search path', function()
+
+        -- Shimmed icon base directories contain the following icons:
+        --
+        --     .icons/icon5.png
+        --     .icons/icon6.xpm
+        --     .icons/icon7.svg
+
+        assert_found_in_path('icon5.png', '/.icons/icon5.png')
+        assert_found_in_path('icon6.xpm', '/.icons/icon6.xpm')
+        assert_found_in_path('icon7.svg', '/.icons/icon7.svg')
+
+        --  Will fail as file does not exist (but extension is supported)
+        assert.is_false(utils.lookup_icon_uncached('icon8.png'))
+    end)
+
+-- Check icon with no (or invalid) extension matching :: if NOT supported_icon_formats[icon_file_ext]
+    it('finds icons without specified path or extension', function()
+
+        -- Shimmed icon base directories contain the following icons:
+        --
+        --     .icons/icon6.xpm
+        --     .icons/icon7.svg
+
+        assert_found_in_path('icon6', '/.icons/icon6.xpm') -- Similar to original tests and testing xpm extension
+        assert_found_in_path('icon7', '/.icons/icon7.svg') -- Similar to original tests and testing svg extension
+
+        assert.is_false(utils.lookup_icon_uncached('/png')) -- [For Uli] supported file does not exist in given location
```

Comment:
`[For Uli]`?

Reply:
Please clean this up, people reading this in the deep future will be confused. After that I will approve/merge. You can use `git commit -a --amend --no-edit && git push --force <origin_name> <branch_name>` to update the commit without creating tons of little commits.